src/item/filter.py

```diff
@@ -228,28 +229,27 @@ def _match_affixes(self, filter_affix_pool: list, item_affix_pool: list[Affix])
 
         for affix in filter_affix_pool:
             if isinstance(affix, dict) and "any_of" in affix:
+                filter_min_affix_count = affix["minAffixCount"] if "minAffixCount" in affix and affix["minAffixCount"] is not None else 1
                 any_of_matched = self._match_affixes(affix["any_of"], item_affix_pool)
-                if len(any_of_matched) > 0:
-                    name = any_of_matched[0]
-                    item_affix_pool = [a for a in item_affix_pool if a.type != name]
-                    matched_affixes.append(name)
+                if len(any_of_matched) >= filter_min_affix_count:
+                    matched_affixes.extend(any_of_matched)
```

Comment:
Should all of any_of_matched be included or rather: `any_of_matched[:filter_min_affix_count]? Usecase: Someone wants two have 2 resistances on the boots. Now they would need to copy the res block twice. With the new implementaiton the goal would be to set minAffixCount: 2 inside the any_of block.

But if we do this also boots with 3 or 4 resistances would show up as matched.

Reply:
Well depends. We can do the following just fine:
```
affixPool:
    - any_of:
      - total_armor
      - damage_reduction_from_close_enemies
      minAffixCount: 2
    - any_of:
      - fire_resistance
      - shadow_resistance
      - poison_resistance
      - lightning_resistance
      - cold_resistance
      minAffixCount: 2
```

Reply:
Imho this defeats the whole point of the any_of block. You might as well just add them outside the any_of. Someone asked about "how can I make certain affixes obligatory". This is currently possible by doing sth like this:

```yaml
BootsTest2:
  itemType: boots
  minPower: 700  
  affixPool:
    - [intelligence]
    - any_of:
      - [movement_speed]
      - [poison_resistance]
      - [cold_resistance]
      - [lightning_resistance]
  minAffixCount: 2
  ```
When we dont remove the additional matches from the any_of block. A boot with 2 light res and cold res but no intelligence will also match the filter. Meaning, there is no difference using the any_of block or not.

Reply:
Maybe we shouldnt call the any_of `minAffixCount` but rather `maxMatches`

Reply:
I had the same idea yesterday. I think the [PoE trade site](https://www.pathofexile.com/trade) filtering is perfect.
The problem here is that every item in affixPool is evaluated using 'and' but minAffixCount makes it an implicit 'or'.

Let me clean this up and provide a different pull request with the PoE logic.